ha/util/health_generator/mock_health_event_publisher.py

```diff
@@ -0,0 +1,217 @@
+_node_id_key = 'node_id'
+_resource_type_key = 'resource_type'
+_resource_id_key = 'resource_id'
+_resource_status_key = 'resource_status'
+_specific_info_key = 'specific_info'
+_delay_key = 'delay'
+
+def is_container_env() -> bool:
+    """Returns True if environment is docker container else False."""
+    docker_env_file_path = pathlib.Path(docker_env_file)
+    if docker_env_file_path.exists():
+        return True
+    return False
+
+def get_data_nodes(conf_store: ConftStoreSearch = None) -> list:
+    """
+    Fetches data node ids using HA wrapper class and displays the result.
+    Args:
+    conf_store: ConftStoreSearch object
+    Returns: list of node ids
+    """
+    data_node_ids = conf_store.get_data_pods(_index)
+    return data_node_ids
+
+def get_server_nodes(conf_store: ConftStoreSearch = None) -> list:
+    """
+    Fetches server node ids using HA wrapper class and displays the result.
+    Args:
+    conf_store: ConftStoreSearch object
+    Returns: list of node ids
+    """
+    server_node_ids = conf_store.get_server_pods(_index)
+    return server_node_ids
+
+def get_disks(args, conf_store: ConftStoreSearch = None) -> None:
+    """
+    Fetches disk ids using ConfStore search API and displays the result.
+    Args:
+    conf_store: ConftStoreSearch object
+    node_id: machine_id value
+    """
+    # TODO: This is temporary code till this is avalable in HA.
+    node_id = args.node_id
+    num_of_cvgs = Conf.get(_index, f'node>{node_id}>storage>num_cvg')
+    disk_ids = []
+    for cvg in range(num_of_cvgs):
+        num_of_data_disks = Conf.get(_index, f'node>{node_id}>storage>cvg[{cvg}]>devices>num_data')
+        num_of_metadata_disks = Conf.get(_index, f'node>{node_id}>storage>cvg[{cvg}]>devices>num_metadata')
+        [ disk_ids.append(Conf.get(_index, f'node>{node_id}>storage>cvg[{cvg}]>devices>data[{dt_disk}]')) for dt_disk in range(num_of_data_disks)]
+        [ disk_ids.append(Conf.get(_index, f'node>{node_id}>storage>cvg[{cvg}]>devices>metadata[{md_disk}]')) for md_disk in range(num_of_metadata_disks)]
+    print(disk_ids)
+
+def get_cvgs(args, conf_store: ConftStoreSearch = None) -> None:
+    """
+    Fetches cvg ids using ConfStore search API and displays the result.
+    Args:
+    conf_store: ConftStoreSearch object
+    node_id: machine_id value
+    """
+    node_id = args.node_id
+    # TODO: This is temporary code till this is avalable in HA.
+    cvg_count = Conf.get(_index, f'node>{node_id}>storage>num_cvg')
+    print([Conf.get(_index, f'node>{node_id}>storage>cvg[{cvg}]>name') for cvg in range(cvg_count)])
+
+def publish(args, conf_store: ConftStoreSearch = None) -> None:
+    """
+    publishes the message on the message bus.
+    Args:
+    conf_store: ConftStoreSearch object
+    node_id: machine_id value
+    config_file: config file path
+    """
+    try:
+        with open(args.file, 'r') as fi:
+            events_dict = json.load(fi)
+            if _events_key in events_dict.keys():
```

Comment:
If multiple events need to be sent, publish function /mock_health_event_publisher.py should be called multiple times right ?

Reply:
Yes, it is called in a for loop.